source/Octopus.Tentacle/Kubernetes/EventMapper.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using k8s.Models;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public record EventRecord(string Reason, string Source, DateTimeOffset OccurredAt);
+    
+    public interface IEventMapper
+    {
+        EventRecord? MapToRecordableEvent(Corev1Event kEvent);
+    }
+
+    public class NfsStaleEventMapper : IEventMapper
+    {
+        public EventRecord? MapToRecordableEvent(Corev1Event kEvent)
+        {
+            const string expectedReason = "NfsWatchdogTimeout";
+            var source = EventHelpers.MetricSourceMapper(kEvent);
+            if (kEvent.Reason.Equals(expectedReason))
+            {
+                var eventTimestamp = EventHelpers.GetLatestTimestampInEvent(kEvent)!.Value.ToUniversalTime();
+                return new EventRecord(expectedReason, source, eventTimestamp);
+            }
+            return null;
+        }
+    }
+    
+    public class AgentKilledEventMapper : IEventMapper
```

Comment:
Maybe `TentacleKilledEventMapper`? I try to think of the Agent as the sum of the parts and the Tentacle is the component we're monitoring in this method.